### ISSUE TITLE
run.py crashed with AttributeError in __init__(): 'unicode' object has no attribute 'iteritems'

### ISSUE THREAD

**Issue by jendrikseipp** [jendrikseipp/rednotebook#246]
RedNotebook crashes at startup.
OS: Ubuntu 12.04 LTS
Version: Last one delivered via Ubuntu-Update

ProblemType: Crash
DistroRelease: Ubuntu 12.04
Package: rednotebook 1.6.1-0~precise1 [origin: LP-PPA-rednotebook-stable]
ProcVersionSignature: Ubuntu 3.2.0-32.51-generic-pae 3.2.30
Uname: Linux 3.2.0-32-generic-pae i686
NonfreeKernelModules: nvidia
ApportVersion: 2.0.1-0ubuntu14
Architecture: i386
CrashCounter: 1
CrashDB: rednotebook
Date: Mon Nov 12 08:52:54 2012
EcryptfsInUse: Yes
ExecutablePath: /usr/share/rednotebook/run.py
InstallationMedia: Ubuntu 12.04 LTS "Precise Pangolin" - Release i386 (20120423.2)
InterpreterPath: /usr/bin/python2.7
PackageArchitecture: all
ProcCmdline: /usr/bin/python /usr/bin/rednotebook
ProcEnviron:
 LANGUAGE=de_AT:de
 PATH=(custom, no user)
 LANG=de_AT.UTF-8
 SHELL=/bin/bash
PythonArgs: ['/usr/bin/rednotebook']
SourcePackage: rednotebook
Title: run.py crashed with AttributeError in __init__(): 'unicode' object has no attribute 'iteritems'
UpgradeStatus: No upgrade log present (probably fresh install)
UserGroups: adm cdrom dip lpadmin plugdev sambashare sudo

**Comment by jendrikseipp** [comment]
*Original comment by* **peter-stadlmaier** (2012-11-12T10:00:28Z):

----------------------------------------

**Comment by jendrikseipp** [comment]
*Original comment by* **jendrikseipp** (2012-11-12T10:34:18Z):

----------------------------------------

It seems like the file /home/pstadl44/Dokumente/RedNote/2011-05.txt is messed up and cannot be read correctly. Have you manually edited it? Can you post the file here? If there's preivate information in it, you can replace it with dummy text and make sure that the error still happens with the "cleaned" file.

**Comment by jendrikseipp** [comment]
*Original comment by* **peter-stadlmaier** (2012-11-12T13:49:38Z):

----------------------------------------

This is the file, which cannot be read correctly.

**Comment by jendrikseipp** [comment]
*Original comment by* **jendrikseipp** (2012-11-12T13:58:25Z):

----------------------------------------

This is not a valid RedNotebook file. Have you edited it manually? 
Here's the relevant section from the manual about the format:


    Data Format

In this paragraph I will explain shortly what the RedNotebook files 
consist of. Firstly it is important to understand that the content is 
saved in a directory with many files, not just one file. The directory 
name is used as a name for the journal.

In the directory there are several files all conforming to the naming 
scheme "2010-05.txt" (<year>-<month>.txt). Obviously these files 
correspond to months (May 2010).

Each month file contains text for the days of that month. The text is 
actuallyYAML <http://www.yaml.org>markup. Without the (unnecessary) 
python directives the files look like this:

24: {text: "This is a normal text entry."}
25:
   Ideas: {"Invent Anti-Hangover machine": null}
   text: "This is another text entry, shown in the main text area."

As you can see the data format uses a dictionary (or hashmap structure) 
for storing the information. The outer dictionary has the day numbers as 
keys and the day content as values. The day values consist of another 
dictionary. It can have a key "text" whose value will be inserted in the 
main content area. Additionally there can be multiple other keys that 
stand for the categories that belong to that day. Each category contains 
a dictionary mapping category entries to the null value.

In summary the data format is a hierarchy of dictionaries. This way the 
format can be easily extended once the need for that arises.

All textual content can be formatted or augmented withtxt2tags 
<http://txt2tags.org/>markup.

**Comment by jendrikseipp** [comment]
*Original comment by* **peter-stadlmaier** (2012-11-12T14:09:20Z):

----------------------------------------

Am 2012-11-12 11:34, schrieb Jendrik Seipp:
> It seems like the file /home/pstadl44/Dokumente/RedNote/2011-05.txt is
> messed up and cannot be read correctly. Have you manually edited it?
No. I also did not (re)move it. It seems do be a valid textfile, which 
can be still opened with an editor.
> Can
> you post the file here?
I uploaded it already on launchpad.

regards,

Peter Stadlmaier

**Comment by jendrikseipp** [comment]
*Original comment by* **jendrikseipp** (2012-12-06T12:54:45Z):

----------------------------------------

I think you'll have to rename the unreadable file and copy-paste the notes from it into RedNotebook.
